### PR TITLE
PullRequest for Issue1715: write clientDataRequst to file

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
AMDS client app controller:
- add the feature to configure the export file name for a scan
- add the feature to export the client data requests to the given file

https://github.com/acquaman/acquaman/issues/1715

### CODE REVIEW

File: source/Connection/AMDSServer.h [hunk 1]
```diff
@@ -20,23 +20,23 @@ class AMDSServer : public QObject
...
-	inline QStringList activeTCPSocketKeys() { return QStringList(activeTCPSockets_.keys()); }
+	inline QStringList activeTCPSocketKeys() const { return QStringList(activeTCPSockets_.keys()); }
 
 	/// helper function to set the list of buffers of the current server
 	void setBufferNames(QStringList activeBufferNames);
 
 signals:
 	/// signal to indicate that something wrong with the connection happened
-	void AMDSServerError(AMDSServer* server, int errorCode, QString socketKey, QString errorMessage);
+	void AMDSServerError(const QString &serverIdentifier, int errorCode, QString socketKey, QString errorMessage);
```
Comment (dretrex) [inline source/Connection/AMDSServer.h]: The sockeyKey and errorMessage should be `const QString &` arguments.

File: source/appController/AMDSClientAppController.h [hunk 2]
```diff
@@ -88,7 +99,12 @@ public slots:
 	void openNetworkSession();
 
 	/// slot to handle socket error signal from the server
-	void onAMDSServerError(AMDSServer* server, int errorCode, const QString &socketKey, const QString &errorMessage);
+	void onAMDSServerError(const QString &serverIdentifier, int errorCode, const QString &socketKey, const QString &errorMessage);
+
+	/// slot to handle the signal to start a new scan
+	void onScanStarted();
+	/// slot to handle the signal to finish a scan
+	void onScanStopped();
```
Comment (davidChevrier) [inline source/appController/AMDSClientAppController.h:108]: Hey @XuetaoLiu-CLS and @dretrex 
Where's the implementation for these functions? I can't seem to find it in the pull request.

Comment (dretrex) [inline source/appController/AMDSClientAppController.h:108]: Don't worry about that.  It is in master, probably came for a ride in one of the other branches.

Comment (dretrex) [inline source/appController/AMDSClientAppController.h:108]: Also, this is the wrong pull request.  Look at #110 instead.

**Pull request merged**